src/main/java/seedu/address/model/room/UniqueRoomList.java

```diff
@@ -0,0 +1,150 @@
+package seedu.address.model.room;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.room.exceptions.DuplicateRoomException;
+import seedu.address.model.room.exceptions.RoomNotFoundException;
+
+/**
+ * A list of rooms that enforces uniqueness between its elements and does not allow nulls.
+ * A room is considered unique by comparing using {@code Room#isSameRoom(Room)}. As such, adding and updating of
+ * rooms uses Room#isSameRoom(Room) for equality so as to ensure that the room being added or updated is
+ * unique in terms of identity in the UniqueRoomList. A room can be removed by passing either its reference or
+ * room number to the remove() method.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Room#isSameRoom(Room)
+ */
+public class UniqueRoomList implements Iterable<Room> {
+
```

Comment:
I presume this is for short-term referencing of a list of rooms (e.g. I want to check all the `SingleRoom`s)

The `RoomList` class should contain a master reference list.

Reply:
This class is a layer of abstraction over the list of `Room`s, similar to `UniquePersonList` in `AddressBook`. Since there is a `UniqueGuestList`, we should maintain consistency and also use `UniqueRoomList`. 

The `RoomList` class is congruent to `AddressBook` in AB4, and thus will contain a `UniqueRoomList`.

We should discuss whether there is a need for `RoomList` class on top of `UniqueRoomList`, and similarly for `GuestList`.